cli/src/rust_template.rs

```diff
@@ -19,20 +18,39 @@ pub enum ProgramTemplate {
     Single,
     /// Program with multiple files for instructions, state...
     Multiple,
+    /// Generate template for Rust unit-test
+    RustTest,
 }
 
 /// Create a program from the given name and template.
-pub fn create_program(name: &str, template: ProgramTemplate) -> Result<()> {
+pub fn create_program(name: &str, template: ProgramTemplate, program_id: &str) -> Result<()> {
     let program_path = Path::new("programs").join(name);
+    let tests = if template == ProgramTemplate::RustTest {
+        "tests"
+    } else {
+        ""
+    };
     let common_files = vec![
-        ("Cargo.toml".into(), workspace_manifest().into()),
+        ("Cargo.toml".into(), workspace_manifest(tests)),
         (program_path.join("Cargo.toml"), cargo_toml(name)),
         (program_path.join("Xargo.toml"), xargo_toml().into()),
     ];
 
     let template_files = match template {
         ProgramTemplate::Single => create_program_template_single(name, &program_path),
         ProgramTemplate::Multiple => create_program_template_multiple(name, &program_path),
+        ProgramTemplate::RustTest => {
+            let mut files = create_program_template_single(name, &program_path);
+            let tests_path = Path::new("tests");
+            files.extend(vec![(
+                tests_path.join("Cargo.toml"),
+                tests_cargo_toml(name),
+            )]);
+            files.extend(create_program_template_rust_test(
+                name, tests_path, program_id,
+            ));
+            files
+        }
```

Comment:
We would want to have Rust tests for both `Single` and `Multiple` program templates. That is going to require some refactoring in order to reduce the duplication if we want to set everything via templates. Refactoring would involve making tests a template as well, `TestTemplate` similar to `ProgramTemplate`.

Refactoring would make it more organized but I think both approaches are fine and it would probably be more convenient for you to add this with your initial approach with `--rust` flag, similar to `--jest` flag.

Reply:
@acheroncrypto 
Thank you for the advice. 
I am currently facing some issues. 
`anchor init --template` option can not take 2 arguments like  `achor init --template single rust-test`. 
In this case, is it acceptable to take like this `achor init --template single --template rust-test`?

Reply:
I think it would be better to separate them as `--program-template` and `--test-template` for clarity instead of using `--template` for both of them.